lib/LanguagePicker.js

```diff
@@ -19,50 +42,83 @@ function LanguagePicker(lang = 'en', config = {}) {
   // Use the given language as first choice
   fallbacks = [lang].concat(fallbacks);
 
-  // Fallback to English
-  if (fallbacks.indexOf('en') === -1) {
-    fallbacks.push('en');
-  }
-
-  // Fallback to nameTag
-  if (this.nameTag) {
-    fallbacks.push(this.nameTag);
-  }
+  // Remove duplicates
+  fallbacks = fallbacks.filter((item, i) => fallbacks.indexOf(item) === i);
 
   // Add prefix to all languages if exists
   // eslint-disable-next-line arrow-body-style
   fallbacks = fallbacks.map((code) => {
     return code === this.nameTag ? code : this.prefix + code;
   });
 
+  // Store initial fallbacks
   this.fallbacks = fallbacks;
 }
 
 LanguagePicker.prototype.newProcessor = function newProcessor() {
-  const { fallbacks, nameTag, prefix } = this;
-  let bestChoiceLang = nameTag || (prefix ? `${prefix}en` : 'en');
-  let bestChoiceValue;
   let firstChoiceValue;
 
   return {
-    addValue(lang, value) {
-      const langIndex = fallbacks.indexOf(lang);
-      if (
-        langIndex > -1 &&
-        langIndex <= fallbacks.indexOf(bestChoiceLang)
-      ) {
-        // Given language is a better fallback
-        bestChoiceLang = lang;
-        bestChoiceValue = value;
+    addValue: (lang, value) => {
+      this.values[lang] = value;
+
+      if (!firstChoiceValue) {
+        firstChoiceValue = value;
+      }
+    },
+    getResult: () => {
+      const valueLangCodes = Object.keys(this.values);
```

Comment:
move this closer to usage - very likely it won't be needed.  (this is not pre-ES5 JS, where vars should be declared at the top. ES5+ is actually opposite - the closer `let` & `const` to the usage, the better)

Reply:
Hmm good point, though declaring at the top is still recommended as far as I can see (also, eslint rule). I can declare it as a 'let' and assign below, though.